data/src/main/java/com/canopas/yourspace/data/storage/database/SenderKeyDao.kt

```
@@ -0,0 +1,58 @@
+    ): SenderKeyEntity?
+
+    suspend fun getSenderKeyRecord(
+        address: String,
+        deviceId: Int,
+        distributionId: String
+    ): SenderKeyRecord? {
+        val entity = getSenderKey(address, deviceId, distributionId)
+        return entity?.let {
+            try {
+                SenderKeyRecord(it.record)
+            } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)